buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/unittest-and-compile.gradle.kts

```diff
@@ -244,3 +245,17 @@ fun removeTeamcityTempProperty() {
 
 val Project.maxParallelForks: Int
     get() = findProperty("maxParallelForks")?.toString()?.toInt() ?: 4
+
+fun failOnEmptySourceFolder() {
+    // Empty source dirs produce cache misses, and are not caught by `git status`.
+    // Fail if we find any.
+    tasks.withType<SourceTask>().configureEach {
```

Comment:
Can we do this check on `buildSrc` as well?

Reply:
Good idea: https://github.com/gradle/gradle/pull/14456